tests/fpgadataflow/test_fpgadataflow_thresholding_binary_search.py

```diff
@@ -0,0 +1,495 @@
+        {"input": 3, "expected_result": 4},
+        {"input": 4, "expected_result": 4},
+        {"input": 7, "expected_result": 8},
+        {"input": 8, "expected_result": 8},
+        {"input": 11, "expected_result": 16},
+        {"input": 15, "expected_result": 16},
+        {"input": 16, "expected_result": 16},
+        {"input": 18, "expected_result": 32},
+        {"input": 27, "expected_result": 32},
+        {"input": 31, "expected_result": 32},
+        {"input": 32, "expected_result": 32},
+        {"input": 42, "expected_result": 64},
+        {"input": 65, "expected_result": 128},
+    ]
+
+    for test_dict in test_vector:
+        output = tbs_inst.find_next_power_of_2(test_dict["input"])
+        assert output >= test_dict["input"]
+        assert output == test_dict["expected_result"]
+
+    return
+
+
+# Test brief: Prove that cppsim is not supported for this class
+@pytest.mark.fpgadataflow
+@pytest.mark.vivado
+def test_fpgadataflow_thresholding_binary_search_cppsim():
+    input_data_type = DataType["UINT16"]
+    act = DataType["BIPOLAR"]
+    fold = -1
+    num_input_channels = 16
+    # 'const' is unsupported see test:
+    # test_fpgadataflow_thresholding_binary_search_const_mem_mode()
+    mem_mode = "decoupled"
+
+    pe = generate_pe_value(fold, num_input_channels)
+    num_steps = act.get_num_possible_values() - 1
+
+    # Generate random, non-decreasing thresholds
+    thresholds = generate_random_threshold_values(
+        input_data_type, num_input_channels, num_steps
+    )
+
+    # make the vivado_hls threshold bug appear (incorrect rtlsim result when first
+    # threshold of first channel is zero, while using BIPOLAR output)
+    if act == DataType["BIPOLAR"]:
+        thresholds[0][0] = 0
+    thresholds = sort_thresholds_increasing(thresholds)
+
+    # Other non-input parameters
+    num_input_vecs = [1, 2, 2]
+    output_data_type = act
+    if output_data_type == DataType["BIPOLAR"]:
+        activation_bias = 0
+    else:
+        activation_bias = output_data_type.min()
+
+    # Generate model from input parameters to the test
+    model = make_single_thresholding_binary_search_modelwrapper(
+        thresholds,
+        pe,
+        input_data_type,
+        output_data_type,
+        activation_bias,
+        mem_mode,
+        num_input_vecs,
+    )
+
+    # Cppsim is not supported for this class, catch the specific exception thrown by
+    # cppsim. Exception raised in cppsim: Custom op_type Thresholding_Binary_Search is
+    # currently not supported.
+    try:
+        model = model.transform(PrepareCppSim())
+        model = model.transform(CompileCppSim())
+        model = model.transform(SetExecMode("cppsim"))
+    except Exception as e:
+        if (
+            str(e)
+            != "Custom op_type Thresholding_Binary_Search is currently not supported."
+        ):
+            raise
+
+
+# Test brief: Prove that memory mode 'const' is not supported for this layer type
```

Comment:
same as above, plus depending on the whole discussion on what const vs decoupled means for this layer

Reply:
removed: https://github.com/Xilinx/finn/pull/687/commits/7f3455fc0d1dafedaf8cdfca8144dea41747a624